pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -839,7 +839,7 @@ func pvAdded(obj interface{}) {
 	// Add VCP-CSI migrated volumes to the volumeIDToNameMap map.
 	// Since cns query will return all the volumes including the migrated ones, the map would need to be a
 	// union of migrated VCP-CSI volumes and CSI volumes, as well.
-	if pv.Spec.CSI == nil && (k8sOrchestratorInstance.IsFSSEnabled(context.Background(), common.CSIMigration) &&
+	if pv.Spec.CSI != nil && (k8sOrchestratorInstance.IsFSSEnabled(context.Background(), common.CSIMigration) &&
```

Comment:
Why are we changing this to **!= nil** here and in pvUpdated and pvDeleted below? 

The case for != nil has already been handled above in line 825.
This is to specifically handle VCP-CSI migrated volumes, for which pv.Spec.CSI == nil.

This block is to handle VCP-CSI migrated volumes  so that  the map will give both migrated volumes and CSI volumes, so that it can match the cnsQuery Result.

Adding @chethanv28 since he has reviewed this part and had suggested the addition on VCP-CSI migrated volumes to the map.

Reply:
In the below lines we are accessing `pv.Spec.CSI.VolumeHandle`, this will cause panic and nil pointer exception. I dont think it is right to access the volumeHandle from `pv.Spec.CSI` when it is nil.

Reply:
pv.Spec.CSI will nil for in-tree migrated volumes

Reply:
we should not change this check - `if pv.Spec.CSI != nil `, as this is for VCP to CSI migration.

Reply:
let me check the code and I will get back to you.

Reply:
@mutgia you have added this code in the PR - https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/1663/files#diff-0e00135fd0f8659d1fd5fa4217c650a97f17c21433ae9076d59b5aadc7d56fc7 

looks like logic is incorrect
we have pv.Spec.CSI == nil and we are accessing `pv.Spec.CSI.VolumeHandle`

We did not face this issue because so far list volume feature was disabled by default.

Reply:
Thanks for confirming. So` pv.Spec.CSI != nil` should remain then. 
 
The check was added based on a comment that the map should be a union of both CSI and VCP-CSI migrated volumes. And I'd followed the code example here: https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/syncer/util.go#L123, which checked for nil, which I'd thought was required for VCP-CSI migrated volumes.